homeassistant/components/melcloud/sensor.py

```diff
@@ -207,9 +203,13 @@ def __init__(
             description.key = f"{description.key}-zone-{zone.zone_index}"
         super().__init__(api, description)
         self._zone = zone
-        self._attr_name = f"{api.name} {zone.name} {description.name}"
 
     @property
-    def native_value(self):
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return f"{self._zone.name} {super().name}"
```

Comment:
Would it be possible to create a device for each room?

Reply:
May be this should be a separate PR?

Reply:
This way of naming an entity (using super().name) with has_entity_name is something we never have anywhere, so I'd rather avoid this kind of setup

Reply:
I think that this should be considered a `breaking change`. The sensors will become unavailable for current device and we will have new sensors for new created device. Are you sure I should go ahead?

Reply:
as long as the unique_id of the sensor doesn't change it's fine

Reply:
Ok, changes implemented, let me know if it looks fine.